app/services/activity/updater.rb

```diff
@@ -173,12 +173,23 @@ def set_commitment
 
       commitment = activity.commitment || Commitment.new
       commitment.activity = activity
-      commitment.value = value
       commitment.transaction_date = infer_transaction_date_from_activity_attributes(activity)
+      convert_and_assign_commitment_value(commitment: commitment, value: value) if value.present?
+
+      return unless commitment.errors.empty?
 
       commitment.save
     end
 
+    def convert_and_assign_commitment_value(commitment:, value:)
+      commitment.value = ConvertFinancialValue.new.convert(value.to_s)
```

Comment:
Is the `value` not always already a string (from the params)?

Reply:
Yeah I think it is, I took this line of code from the other places we use `ConvertFinancialValue` but I think those cover the CSV upload as well, which might be why they do that conversion. I'll remove this as it's not necessary.